trantor/unittests/InetAddressUnittest.cc

```diff
@@ -11,6 +11,9 @@ TEST(InetAddress, innerIpTest)
     EXPECT_EQ(true, InetAddress("10.0.0.1", 0).isIntranetIp());
     EXPECT_EQ(true, InetAddress("172.31.10.1", 0).isIntranetIp());
     EXPECT_EQ(true, InetAddress("127.0.0.1", 0).isIntranetIp());
+    EXPECT_EQ(true, InetAddress("example.com", 0).isUnspecified());
+    EXPECT_EQ(false, InetAddress("127.0.0.2", 0).isUnspecified());
```

Comment:
This is wrong. The range is `127.0.0.0/8`

Reply:
Unspecified means `InetAddress` failed to parse the supplied address. Akin to how ASIO works.

Reply:
IMHO it doens't matter what ASIO does.
[RFC 1122 Section 3.2.1.3 g](https://tools.ietf.org/html/rfc1122#section-3.2.1.3) defines what a loopback adress looks like so the RFC should be the reference not any Software implementation.

Reply:
I think Rust did a pretty good job ob describing and implemeting ipv4 and ipv6
https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html
https://doc.rust-lang.org/std/net/struct.Ipv6Addr.html